action.yaml

```diff
@@ -72,17 +82,43 @@ runs:
           PRE_COMMIT_ARGS="--files ${{ steps.changed-files.outputs.all_changed_files }}"
           echo "PRE_COMMIT_ARGS=$PRE_COMMIT_ARGS" >> "$GITHUB_ENV"
         fi
+
+    - name: Compute pre-commit cache key
+      if: inputs.s3-bucket-name != ''
+      id: pre_commit_cache_key
+      run: |
+        echo "key=${{ hashFiles('**/.pre-commit-config.yaml') }}" >> $GITHUB_OUTPUT
+      shell: bash
+    - name: Load cached pre-commit hooks if available
+      if: inputs.s3-bucket-name != ''
+      uses: everpcpc/actions-cache@v2
+      id: read_cache
```

Comment:
do we also need a write_cache action after running pre-commit in case the read was a miss?